File: app/src/main/java/io/github/javiewer/activity/SecureActivity.java
package io.github.javiewer.activity;

import androidx.appcompat.app.AppCompatActivity;
import android.view.WindowManager;

/**
 * SecureActivity
 *
 * @author lao
 * @date 2020/9/4
 * Profile: 基础Activity
 */
public class SecureActivity extends AppCompatActivity {

    @Override
    protected void onPause() {
        super.onPause();
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_SECURE, WindowManager.LayoutParams.FLAG_SECURE);
    }

    @Override
    protected void onResume() {
        super.onResume();
        getWindow().clearFlags(WindowManager.LayoutParams.FLAG_SECURE);
    }
}

File: app/src/main/java/io/github/javiewer/activity/SecureActivity.kt
package io.github.javiewer.activity

import android.view.WindowManager
import androidx.appcompat.app.AppCompatActivity

/**
 * 安全 Activity 基类，在暂停时设置 FLAG_SECURE 防止截屏。
 *
 * 所有需要防截屏保护的 Activity 应继承此类。
 * 恢复时自动清除 FLAG_SECURE 以允许正常截屏。
 */
open class SecureActivity : AppCompatActivity() {

    override fun onPause() {
        super.onPause()
        window.setFlags(
            WindowManager.LayoutParams.FLAG_SECURE,
            WindowManager.LayoutParams.FLAG_SECURE
        )
    }

    override fun onResume() {
        super.onResume()
        window.clearFlags(WindowManager.LayoutParams.FLAG_SECURE)
    }
}
